client/my-sites/checkout/composite-checkout/hooks/use-valid-checkout-back-url.ts

```diff
@@ -4,8 +4,13 @@ import { useSelector } from 'react-redux';
 import { resemblesUrl } from 'calypso/lib/url';
 import getInitialQueryArguments from 'calypso/state/selectors/get-initial-query-arguments';
 
-const getAllowedHosts = ( siteSlug: string ) => {
-	const basicHosts = [ 'jetpack.com', 'jetpack.cloud.localhost', 'cloud.jetpack.com', siteSlug ];
+const getAllowedHosts = ( siteSlug?: string ) => {
```

Comment:
In some cases, (like when the user is logged out), the`siteSlug` will be undefined and we need to allow for that.